cmd/kubeadm/app/cmd/phases/preflight.go

```diff
@@ -56,6 +55,52 @@ var (
 	errorMissingConfigFlag = errors.New("the --config flag is mandatory")
 )
 
+// preflightMasterContext defines the behavior that a runtime context passed to the PreflightMaster master phase
+// should have. Please note that we are using an interface in order to make this phase reusable in different workflows
+// (and thus with different runtime context, all of them requested to be compliant to this interface)
+type preflightMasterContext interface {
+	Cfg() *kubeadmapi.InitConfiguration
+	DryRun() bool
+	IgnorePreflightErrors() sets.String
+}
+
+// PreflightMaster defines that kubeadm workflow phase that implements preflight checks for a new master node.
+var PreflightMaster = workflow.SimplePhase{
```

Comment:
probably better to make such phase variables private and have a public getter instead.
`GetPreflightMaster() { return preflightMaster }`

otherwise any package can modify the PreflightMaster variable.
also we might want to prefix these with Phase - e.g.
`PhasePreflightMaster` ?